### ISSUE TITLE
Getting invalid argument error with 7.7 and 7.7.1

### ISSUE THREAD

**Issue by johnfleming** [WordPress/gutenberg#20907]
**Describe the bug**
Using Gutenberg 7.7.1 on site, kindneyindiana.org with Editors Kit plugin.  Getting error on Editors Kit landing page when Gutenberg active (Don't get it with just 5.3.2

Invalid Argument supplied for foreach() in /home/customer/www/kidneyindiana.org/public_html/wp-content/plugins/gutenberg/lib/client-assets.php on line 619

Contacted developer. He asked me to report here since Gutenberg throwing the error

**To reproduce**
1. Go to admin
2. Click on 'Setting, Editors Kit
3. See error

**Expected behavior**
No error

**Screenshots**
![2020-03-14_15-24-24](https://user-images.githubusercontent.com/10209456/76688931-f5087d80-6607-11ea-8230-c81d836067a2.png)

**Desktop (please complete the following information):**
 - OS: [OSX Catalina]
 - Browser [Chroms
 - Version [Version 80.0.3987.132]

**Smartphone (please complete the following information):**

**Additional context**
- Please add the version of Gutenberg you are using in the description. 7.7.1
- To report a security issue, please visit the WordPress HackerOne program: https://hackerone.com/wordpress.

**Comment by aduth** [comment]
After some initial debugging, I traced it back to this code:

https://github.com/WordPress/gutenberg/blob/8905f96abd3b86765f5894f940ccf0c9e9c331c1/lib/client-assets.php#L638-L643

This was added as part of #17151 (cc @ryelle).

It’s a feature which will be included in the upcoming WordPress 5.4 release. This specific code was backported as part of the commit at https://github.com/WordPress/wordpress-develop/commit/437ac63d7652b08095dc7f77b6ba4ef52db44dbb .

I believe the warning may only affect the plugin, based on how the message is about the foreach argument being invalid. In the core adaptation of the code, `$available_image_sizes` is guaranteed to be an array ([source](https://github.com/WordPress/wordpress-develop/blob/c8d918e47634bd423e0c7ccfc0e8f7a34515c3f5/src/wp-admin/edit-form-blocks.php#L222)). In the plugin, I’m not sure `$settings['imageSizes']` is safe to assume as such (thus causing the warning).

Action items:

- [ ] Test the combination of this plugin with [WordPress 5.4.0 RC 2](https://wordpress.org/news/2020/03/wordpress-5-4-rc2/) to verify the above assumptions about it only affecting the plugin.
- [ ] Fix the plugin code above to be more durable about the values it is expecting to iterate in the `foreach` loop. It's quite possible it's assuming some code which only exists in 5.4.x. Since the plugin supports WordPress 5.3.x, it should be expected to work there as well.

**Comment by aduth** [comment]
I've confirmed this issue is not present with EditorsKit + WordPress 5.4 RC 2.

Based on the steps to reproduce, I was able to track this back to an incompatibility with this line of code in EditorsKit welcome screen:

https://github.com/phpbits/block-options/blob/8a018909bb7407719afc23282a7303e2f5eb5357/includes/class-editorskit-welcome.php#L126

Since EditorsKit is running the `block_editor_settings` filter with an empty array, Gutenberg's subsequent iteration of `foreach ( $settings['imageSizes'] )` will fail.

There's two options here to address this:

1. EditorsKit could provide all of the [expected settings properties](https://github.com/WordPress/wordpress-develop/blob/c8d918e47634bd423e0c7ccfc0e8f7a34515c3f5/src/wp-admin/edit-form-blocks.php#L286-L313), to maximize compatibility with filters which may assume those properties to be present.
2. As noted in https://github.com/WordPress/gutenberg/issues/20907#issuecomment-599740671, Gutenberg's filtering should be made to be more durable about the inputs it works with.
   1. Don't assume that `imageSizes` is an array.
   2. Avoid assigning `imageDimensions` if already assigned (?), if the idea is that this code need only exist as compatibility for 5.3.2 (it should likely be moved to `compat.php`, with comments indicating some future removal once minimum version requirement is bumped).
